pkg/resources/secret_common.go

```diff
@@ -0,0 +1,112 @@
+)
+
+var secretCommonSchema = map[string]*schema.Schema{
+	"name": {
+		Type:             schema.TypeString,
+		Required:         true,
+		Description:      blocklistedCharactersFieldDescription("String that specifies the identifier (i.e. name) for the secret, must be unique in your schema."),
+		DiffSuppressFunc: suppressIdentifierQuoting,
+	},
+	"database": {
+		Type:             schema.TypeString,
+		Required:         true,
+		Description:      blocklistedCharactersFieldDescription("The database in which to create the secret"),
+		ForceNew:         true,
+		DiffSuppressFunc: suppressIdentifierQuoting,
+	},
+	"schema": {
+		Type:             schema.TypeString,
+		Required:         true,
+		Description:      blocklistedCharactersFieldDescription("The schema in which to create the secret."),
+		ForceNew:         true,
+		DiffSuppressFunc: suppressIdentifierQuoting,
+	},
+	"comment": {
+		Type:        schema.TypeString,
+		Optional:    true,
+		Description: "Specifies a comment for the secret.",
+	},
+	ShowOutputAttributeName: {
+		Type:        schema.TypeList,
+		Computed:    true,
+		Description: "Outputs the result of `SHOW SECRET` for the given secret.",
+		Elem: &schema.Resource{
+			Schema: schemas.ShowSecretSchema,
+		},
+	},
+	DescribeOutputAttributeName: {
+		Type:        schema.TypeList,
+		Computed:    true,
+		Description: "Outputs the result of `DESCRIBE SECRET` for the given secret.",
+		Elem: &schema.Resource{
+			Schema: schemas.DescribeSecretSchema,
+		},
+	},
+	FullyQualifiedNameAttributeName: schemas.FullyQualifiedNameSchema,
+}
+
+type commonSecretCreate struct {
+	name     string
+	database string
+	schema   string
+	comment  *string
+}
+
+func handleSecretCreate(d *schema.ResourceData) commonSecretCreate {
+	create := commonSecretCreate{
+		name:     d.Get("name").(string),
+		database: d.Get("database").(string),
+		schema:   d.Get("schema").(string),
+	}
+	if v, ok := d.GetOk("comment"); ok {
+		create.comment = sdk.Pointer(v.(string))
+	}
+
+	return create
+}
+
+func handleSecretRead(d *schema.ResourceData, id sdk.SchemaObjectIdentifier, secret *sdk.Secret, secretDescription *sdk.SecretDetails) error {
+	if err := d.Set(FullyQualifiedNameAttributeName, id.FullyQualifiedName()); err != nil {
+		return err
+	}
+	if err := d.Set("name", id.Name()); err != nil {
+		return err
+	}
+	if err := d.Set("database", secret.DatabaseName); err != nil {
+		return err
+	}
+	if err := d.Set("schema", secret.SchemaName); err != nil {
+		return err
+	}
+	if err := d.Set("comment", secret.Comment); err != nil {
+		return err
+	}
+	if err := d.Set(ShowOutputAttributeName, []map[string]any{schemas.SecretToSchema(secret)}); err != nil {
+		return err
+	}
+	if err := d.Set(DescribeOutputAttributeName, []map[string]any{schemas.SecretDescriptionToSchema(*secretDescription)}); err != nil {
+		return err
+	}
+	return nil
+}
+
+func handleSecretUpdate(id sdk.SchemaObjectIdentifier, d *schema.ResourceData) *sdk.AlterSecretRequest {
```

Comment:
I would refactor this function to take in `*sdk.AlterSecretRequest` rather than create it here and either return it or nil. I can imagine there's a bug (or could be) that we would like to use that request and it's nil, because there were no changes done to "comment" field where the request is created.